pkg/compat/client.go

```diff
@@ -171,7 +196,7 @@ func (c client) downConvert(ctx context.Context, obj runtime.Object) (runtime.Ob
 
 //
 // upConvert will convert src resource to dst as needed based on cluster version
-func (c client) upConvert(ctx context.Context, src runtime.Object, dst runtime.Object) error {
+func (c client) upConvert(ctx context.Context, src k8sclient.Object, dst k8sclient.Object) error {
```

Comment:
@sseago Is `k8sclient.Object` a good substitute for `runtime.Object`?

Reply:
It's "good" in the sense that all of the controller-runtime API calls have replaced one with the other, so we need to do the same. Basically client.Object includes both an inline reference to runtime.Object *and* an online meta reference, so it contains more information than what was passed in the older version.